xapian-core/tests/perftest/perftest.h

```diff
@@ -95,6 +98,23 @@ class PerfTestLogger {
      */
     void searching_end();
 
+    /** Log the start of a diversifcation run.
+     */
+    void diversifying_start(const std::string & description);
+
+    /** Log the start of diversification.
+     */
+    void diversify_start();
+
+    /** Log the completion of diversification.
+     */
+    void diversify_end(const unsigned int k,
```

Comment:
I'm not sure there's a good reason to make `k` const here - it's passed by value.

Reply:
I agree that ```k``` is passed by value so a ```const``` here is redundant. But after having looked at [this](https://stackoverflow.com/questions/8714250/isnt-const-redundant-when-passing-by-value), I thought it would be a better practice to disallow modifications of ```k``` locally inside ```diversify_end``` as it's going to be only logged.

Reply:
There's no good reason to mark it `const` here, even if you mark it `const` in the definition (for a pass-by-value parameter the `const`-ness doesn't need to match between declaration and definition).  It's not part of the API that it's unchanged inside the method - that makes no difference to the calling code.

I probably wouldn't bother on the definition myself.  I see the argument that it can help the reader comprehend a large function, but this one is only 4 statements/10 lines.

Reply:
Thanks for clarifying this up. Modification pushed in a fixup commit.